src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -1973,6 +1973,22 @@ internal static void UpdateExceptionErrorRecordPosition(Exception exception, ISc
                 }
             }
         }
+
+        internal static void UpdateExceptionErrorRecordHistoryId(RuntimeException exception, ExecutionContext context)
+        {
+            InvocationInfo invInfo = exception.ErrorRecord.InvocationInfo;
+            if (invInfo is null || invInfo.HistoryId != -1)
```

Comment:
```suggestion
            if (invInfo is not { HistoryId: -1 })
```

Subjectively clearer new syntax, feel absolutely free to just mark as resolved if you disagree (it does compile the same FYI)